pkg/auth/authz_handler.go

```diff
@@ -105,3 +101,44 @@ func OrgIDFromContext(ctx context.Context) string {
 	payload := PayloadFromContext(ctx)
 	return payload.Organization
 }
+
+func (a *AuthzHandler) isAuthorizedToAccessRoute(
+	request *http.Request,
+	payload *ocm.AuthPayload) bool {
+
+	route := middleware.MatchedRouteFrom(request)
+
+	allScopesAreAllowedResponse := func() bool {
+		a.log.Debugf(
+			"%s: Authorized user: %s all scopes are allowed",
+			route.PathPattern, payload.Username)
+		return true
+	}
+
+	if route.Authenticators == nil {
+		return allScopesAreAllowedResponse()
+	}
+
+	authScheme := route.Authenticator.Schemes[0]
+	for _, policy := range route.Authenticators {
+		policyScopes, exist := policy.Scopes[authScheme]
+		if !exist {
+			continue
+		}
+		if len(policyScopes) == 0 {
+			return allScopesAreAllowedResponse()
```

Comment:
don't we need to also `continue` in this case?

Reply:
If we get to this line it means the route is defined with an empty list for the required auth.
Currently it can only happen only with agentAuth, in such cases all scopes are allowed.